proposals/xxxx-mesh-nodes.md

```diff
@@ -0,0 +1,329 @@
+
+The mesh node entry metadata supports two signature definitions:
+- The Output signature describes `vertices` output elements
+- The PatchConstantOrPrimitive signature describes `primitives` output elements.
+
+In addition to the intrinsics supported by a broadcast launch node shader with
+no output nodes, the following mesh shader intrinsics are also supported by
+node shaders with `mesh` launch type:
+- `dx.op.setMeshOutputCounts`
+- `dx.op.EmitIndices`
+- `dx.op.StoreVertexOutput`
+- `dx.op.StorePrimitiveOutput`
+
+Notably, the `dx.op.GetMeshPayload` intrinsic from mesh shader is not supported
+by mesh nodes. Instead, payload data is available through the node input record.
+
+### Runtime Additions
+
+#### Runtime information
+
+For mesh launch mode, additional node details will need to be captured to RDAT.
+For context, see [RDAT_LibraryTypes.inl](https://github.com/microsoft/DirectXShaderCompiler/blob/main/include/dxc/DxilContainer/RDAT_LibraryTypes.inl).
+The approach proposed for this prototype is to add a `RDAT::NodeFuncAttribKind`
+of `MeshShaderInfo`, and add an entry to the union:
+`RDAT_RECORD_REF(MSInfo, MeshShaderInfo)`.
+This will use the existing `MSInfo` structure to encode the required details
+for the mesh shader.
+There are a few redundant fields of this structure, given what's already
+defined for the node shader, but for the experimental feature, the trade-off is
+probably worth the simplicity of not adding another record table at this point.
+
+For now, here is a breakdown of the fields in the `MSInfo` structure, and how
+they will be used in a mesh node:
+- `SigOutputElements` - used for the elements of the `vertices` output array.
+- `SigPrimOutputElements` - used for the elements of the `primitives` output array.
+- `ViewIDOutputMask` - unset for now.
+- `ViewIDPrimOutputMask` - unset for now.
+- `NumThreads` - redundant with `NodeShaderFuncAttrib` NumThreads.  Could be used instead of the additional attribute, or we could copy the ref so it's identical in both places.
+- `GroupSharedBytesUsed` - redundant with same field in `NodeShaderInfo`.  Set to same value.
+- `GroupSharedBytesDependentOnViewID` - set to zero for now.
+- `PayloadSizeInBytes` - redundant with input record size info.  We shouldn't use this one.  Set to zero?
+- `MaxOutputVertices` - based on `vertices` array size
+- `MaxOutputPrimitives` - based on `primitives` array size
+- `MeshOutputTopology` - `uint8_t` encoded `DXIL::MeshOutputTopology` based on `[OutputTopology(`*topology*`)]`
+
+#### Device Capability
+
+See the work graphs spec
+[here](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#graphics-nodes)
+for detail.
+
+Devices that support experimental `D3D_SHADER_MODEL_6_9` and experimental [`D3D12_WORK_GRAPHS_TIER_1_1`](https://github.com/microsoft/DirectX-Specs/blob/master/d3d/WorkGraphs.md#d3d12_work_graphs_tier) are required to support these features as part of graphics nodes in work graphs.
```

Comment:
Presumably mesh shaders would need to be supported too?

Reply:
Apps don't have to check for mesh shader support, as it's implicitly a requirement for work graphs tier 1.1.  You could mention it that way (I'll also clarify in work graphs spec).